src/mjolnir.py

```diff
@@ -118,9 +118,17 @@ def _on_collect_status(self, event: ops.CollectStatusEvent) -> None:
         if not container.can_connect():
             self._charm.unit.status = ops.MaintenanceStatus("Waiting for pebble")
             return
-        service = container.get_services(MJOLNIR_SERVICE_NAME)
-        if service:
-            logger.debug("%s service already exists, skipping")
+        mjolnir_service = container.get_services(MJOLNIR_SERVICE_NAME)
+        if mjolnir_service:
+            logger.debug("%s service already exists, skipping", MJOLNIR_SERVICE_NAME)
+            return
+        current_services = container.get_services()
+        all_svcs_running = all(svc.is_running() for svc in current_services.values())
+        if not all_svcs_running or not current_services:
+            # The get_membership_room_id does a call to Synapse API in order to get the
+            # membership room id. This only works if Synapse and NGINX are running so that's why
```

Comment:
I think NGINX has to be checked using the NGINX container ?

Reply:
Wow...excellent catch! I totally miss that.

Well, let's see:
One: my mistake, we don't need the NGINX container now since the API calls are done directly to Synapse.
Two: it's an edge case but is possible that the container has one or more services running but not the one that we need, the Synapse service.

I think this solves even the readability of this check because it could be changed only to check specifically the Synapse service.
What do you think?